spot-contracts/contracts/strategies/FeeStrategy.sol

```diff
@@ -0,0 +1,227 @@
+        require(targetDeviation_ > 0, "FeeStrategy: target deviation too low");
+        targetDeviation = targetDeviation_;
+    }
+
+    /// @notice Updates the mint fee parameters.
+    /// @param maxMintFeePerc_ The new mint fee ceiling percentage
+    ///        as a fixed point number with {DECIMALS} places.
+    function updateMintFees(uint256 maxMintFeePerc_) external onlyOwner {
+        require(maxMintFeePerc_ <= ONE, "FeeStrategy: mint fee too high");
+        maxMintFeePerc = maxMintFeePerc_;
+    }
+
+    /// @notice Updates the burn fee parameters.
+    /// @param maxBurnFeePerc_ The new burn fee ceiling percentage
+    ///        as a fixed point number with {DECIMALS} places.
+    function updateBurnFees(uint256 maxBurnFeePerc_) external onlyOwner {
+        require(maxBurnFeePerc_ <= ONE, "FeeStrategy: burn fee too high");
+        maxBurnFeePerc = maxBurnFeePerc_;
+    }
+
+    /// @notice Update the parameters determining the slope and asymptotes of the sigmoid fee curve.
+    /// @param p Lower, Upper and Growth sigmoid paramters are fixed point numbers with {DECIMALS} places.
+    function updateRolloverFees(SigmoidParams calldata p) external onlyOwner {
+        require(p.lower >= -int256(SIGMOID_BOUND), "FeeStrategy: fee lower bound too low");
+        require(p.upper <= int256(SIGMOID_BOUND), "FeeStrategy: fee upper bound too high");
+        require(p.lower <= p.upper, "FeeStrategy: paramters invalid");
+        rolloverFeeAPR.lower = p.lower;
+        rolloverFeeAPR.upper = p.upper;
+        rolloverFeeAPR.growth = p.growth;
+    }
+
+    /// @inheritdoc IFeeStrategy
+    function computeMintFeePerc() external override returns (uint256) {
+        // NOTE: when vaultTVL > targetTVL, we want to encourage supply growth;
+        //       we thus drop mint fees to 0
+        return (computeNormalizedDeviation(perp.getDepositBond()) > ONE) ? 0 : maxMintFeePerc;
+    }
+
+    /// @inheritdoc IFeeStrategy
+    function computeBurnFeePerc() external override returns (uint256) {
+        // NOTE: when vaultTVL < targetTVL, we want to encourage supply reduction;
+        //       we thus drop burn fees to 0
+        return (computeNormalizedDeviation(perp.getDepositBond()) < ONE) ? 0 : maxBurnFeePerc;
+    }
+
+    /// @inheritdoc IFeeStrategy
+    function computeRolloverFeePerc() external override returns (int256) {
+        IBondController referenceBond = perp.getDepositBond();
+
+        int256 rolloverAPR = Sigmoid.compute(
+            computeNormalizedDeviation(referenceBond).toInt256(),
+            rolloverFeeAPR.lower,
+            rolloverFeeAPR.upper,
+            rolloverFeeAPR.growth,
+            ONE.toInt256()
+        );
+
+        // We calculate the rollover fee for the given cycle by dividing the annualized rate
+        // by the number of cycles in any given year.
+        return (rolloverAPR * referenceBond.duration().toInt256()) / ONE_YEAR_SEC;
```

Comment:
```suggestion
        // TODO: use the bond duration from the bondIssuer rather than the deposit bond
        return (rolloverAPR * referenceBond.duration().toInt256()) / ONE_YEAR_SEC;
```